safenode/src/protocol/types/wallet/mod.rs

```diff
@@ -0,0 +1,109 @@
+//!     b. Fetch the register with address of either the plaintext of or the encrypted `PublicAddress`.
+//!     c. Decrypt all entries and apply the ops to your Wallet, to get the current state of it.
+//!     d. If there is another register linked at the end of this one, follow that link and repeat steps b., c. and d.
+//!
+//! We will already now pave for that, by mimicing that flow for the local storage of a Wallet.
+//! First though, a simpler local storage will be used. But after that a local register store can be implemented.
+
+mod error;
+mod local_store;
+mod network_store;
+
+pub use self::{
+    error::{Error, Result},
+    local_store::{LocalDepositWallet, LocalSendWallet},
+    // network_store::NetworkWallet,
+};
+
+use super::transfers::{CreatedDbc, Outputs as TransferDetails};
+
+use sn_dbc::{Dbc, DbcIdSource, DerivedKey, MainKey, PublicAddress, Token};
+
+use async_trait::async_trait;
+
+/// A SendClient is used to transfer tokens to other addresses.
+///
+/// It does so by creating a transfer and returning that to the caller.
+/// It is expected that the implementation of this trait is a network client,
+/// that will also upload the transfer to the network before returning it.
+/// The network will validate the transfer upon receiving it. Once enough peers have accepted it,
+/// the transfer is completed.
+///  
+/// For tests the implementation can be a local client with no network connection,
+/// that will just return the transfer to the caller.
+#[async_trait]
+pub trait SendClient {
+    ///
+    async fn send(
+        &self,
+        dbcs: Vec<(Dbc, DerivedKey)>,
+        to: Vec<(Token, DbcIdSource)>,
+        change_to: PublicAddress,
+    ) -> Result<TransferDetails>;
+}
+
+/// A send wallet is a wallet that can send tokens to other addresses.
+/// It is also a deposit wallet, so it can receive tokens from other wallets.
+#[async_trait]
+pub trait SendWallet<C: SendClient> {
+    // /// Loads a wallet from the given path.
+    // fn load_from(path: &Path) -> Self;
+    /// Creates a new wallet with the given key.
+    fn new(key: MainKey, client: C) -> Self;
+    /// The address of the wallet, to which others send tokens.
+    fn address(&self) -> PublicAddress;
+    /// The current balance of the wallet.
+    fn balance(&self) -> Token;
+    /// Used to generate a new dbc id for receiving tokens.
+    fn new_dbc_address(&self) -> DbcIdSource;
+    /// Will only deposit those that are actually accessible by this wallet.
+    fn deposit(&mut self, dbcs: Vec<Dbc>);
+    /// Sends the given tokens to the given addresses.
+    /// Returns the new dbcs that were created.
+    /// Depending on the implementation of the send client, this may
+    /// also register the transaction with the network.
+    async fn send(&mut self, to: Vec<(Token, PublicAddress)>) -> Result<Vec<CreatedDbc>>;
+}
+
+/// A deposit wallet is a wallet that can receive tokens from other wallets.
+/// It can however not send tokens to other addresses.
```

Comment:
If SendWallet can do the both, then why have a separate DepositWallet ?

Reply:
Because you might want a wallet which can't send.

For example, the nodes only receiving rewards don't need the api to send tokens.

They can instantiate such wallets outside of the node software. Such as in a client.